src/commons/sideContent/SideContentReducer.ts

```diff
@@ -1,72 +1,87 @@
+import { createReducer } from '@reduxjs/toolkit';
+import { Reducer } from 'redux';
+
 import { defaultSideContent, defaultSideContentManager } from '../application/ApplicationTypes';
 import { SourceActionType } from '../utils/ActionsHelper';
-import { getDynamicTabs, getTabId } from './SideContentHelper';
-import { getLocation } from './SideContentHelper';
-import { CHANGE_SIDE_CONTENT_HEIGHT } from './SideContentTypes';
+import { changeSideContentHeight, endAlertSideContent } from './SideContentActions';
+import { getDynamicTabs, getLocation, getTabId } from './SideContentHelper';
 import {
-  END_ALERT_SIDE_CONTENT,
   REMOVE_SIDE_CONTENT_ALERT,
   RESET_SIDE_CONTENT,
   SideContentManagerState,
   SPAWN_SIDE_CONTENT,
   VISIT_SIDE_CONTENT
 } from './SideContentTypes';
 
-export function SideContentReducer(
-  state: SideContentManagerState = defaultSideContentManager,
+export const SideContentReducer: Reducer<SideContentManagerState> = (
+  state = defaultSideContentManager,
   action: SourceActionType
-): SideContentManagerState {
+) => {
   if (!(action as any).payload?.workspaceLocation) {
     return state;
   }
+  state = storySideContentReducer(state, action);
+  state = nonStorySideContentReducer(state, action);
+  state = oldSideContentReducer(state, action);
+  return state;
+};
+
+const storySideContentReducer: Reducer<SideContentManagerState> = (
+  state = defaultSideContentManager,
+  action
+) => {
+  const [workspaceLocation, storyEnv] = getLocation((action as any).payload.workspaceLocation);
+  if (workspaceLocation !== 'stories') {
+    return state;
+  }
+
+  const sideContentState = state.stories[storyEnv];
+  return createReducer(defaultSideContentManager, builder => {
+    builder
+      .addCase(changeSideContentHeight, (state, action) => {
+        state.stories[storyEnv].height = action.payload.height;
```

Comment:
Note that `sideContentState` above is outside the `createReducer` function; thus it is not a workable draft state from Immer and should not be written to, i.e. `sideContentState.XXX = YYY` is not allowed.